services/worker-manager/src/providers/azure.js

```diff
@@ -621,6 +632,8 @@ class AzureProvider extends Provider {
       if (!worker.providerData.vm.id) {
         return worker.state;
       }
+      // XXX note that this doesn't actually change the state to RUNNING
+      // (that happens in registerWorker)
       return this.Worker.states.RUNNING;
```

Comment:
Yikes, this comment is correct and we shouldn't be doing this here. We should probably outright return `worker.state` and refactor to check a `providerData` field that indicates the worker's current state, i.e. at this point we are in state "all resources are provisioned but not yet registered"

Reply:
Yeah, that can happen in one of the followups I made a card for in https://github.com/taskcluster/taskcluster/projects/7.  The "XXX" is just to highlight that this is surprising :)